src/sym.cpp

```diff
@@ -245,28 +254,31 @@ std::vector<std::string> SymbolTable::closestTypeMatch(const char *str, bool str
     // maxDelta, return the first non-empty vector of one or more sets of
     // alternatives with minimal edit distance.
     const int maxDelta = 2;
-    std::vector<std::string> matches[maxDelta + 1];
-
-    TypeMapType::const_iterator iter;
-    for (iter = types.begin(); iter != types.end(); ++iter) {
-        // Skip over either StructTypes or EnumTypes, depending on the
-        // value of the structsVsEnums parameter
-        bool isEnum = (CastType<EnumType>(iter->second) != NULL);
-        if (isEnum && structsVsEnums)
-            continue;
-        else if (!isEnum && !structsVsEnums)
-            continue;
-
-        int dist = StringEditDistance(str, iter->first, maxDelta + 1);
-        if (dist <= maxDelta)
-            matches[dist].push_back(iter->first);
+    std::array<std::vector<std::string>, maxDelta + 1> matches;
+
+    for (const TypeMapType &typeMap : types) {
```

Comment:
This is in `closestTypeMatch` and doesn't seem to me like the order matters.

Reply:
Hmmm, this can be invoked only through `SymbolTable::ClosestEnumTypeMatch()`. While `SymbolTable::ClosestTypeMatch()` seems to be completely unused.

And trying to redefine enum in the nested scope in ISPC I see that it fails like it was not designed for that...

Reply:
Either way, since the function is only checking for the edit distance, I think it's okay to leave it. If you'd like me to change it anyway or for a different reason, I can still do that.